File: src/test/java/aoc2015/Day01Test.java
package aoc2015;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class Day01Test {

    @Test
    public void part1Test() {
        var day = new Day01();

        assertEquals("0", day.part1("(())"));
        assertEquals("3", day.part1("(()(()("));
        assertEquals("3", day.part1("))((((("));
        assertEquals("-1", day.part1("))("));
        assertEquals("-3", day.part1(")())())"));
    }

    @Test
    public void part2Test() {
        var day = new Day01();

        assertEquals("1", day.part2(")"));
        assertEquals("5", day.part2("()())"));
    }
}

File: aoc25/src/test/java/day01/Day01Test.kt
package day01

import aoc.loadExample
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class Day01Test {

    @Test
    fun part1Example() {
        val input = loadExample("day01.txt")
        val subject = Day01()

        val result = subject.part1(input)

        assertThat(result).isEqualTo(3)
    }

    @Test
    fun part2Example() {
        val input = loadExample("day01.txt")
        val subject = Day01()

        val result = subject.part2(input)

        assertThat(result).isEqualTo(6)
    }

    @Test
    fun part2BigRotationCountsMultipleTimes() {
        val input = "R1000"
        val subject = Day01()

        val result = subject.part2(input)

        assertThat(result).isEqualTo(10)
    }
}